docs/preprocessor/rewrite-links.py

```
@@ -0,0 +1,316 @@
+
+
+def build_codeberg_url(base_url: str, branch: str, repo_path: Path, suffix: str) -> str:
+    encoded_path = quote(repo_path.as_posix(), safe="/")
+    return f"{base_url.rstrip('/')}/src/branch/{quote(branch, safe='')}/{encoded_path}{suffix}"
+
+
+def rewrite_target(
+    target: str,
+    chapter_path: str,
+    docs_root: Path,
+    src_dir: str,
```

Ruff (0.15.15)

[warning] Unused function argument: `src_dir`

(ARG001)